autoload/ale/lsp/window.vim

```diff
@@ -0,0 +1,83 @@
+" Author: suoto <andre820@gmail.com>
+" Description: Handling of window/* LSP methods, although right now only
+" handles window/showMessage
+
+" Constants for message type codes
+let s:LSP_MESSAGE_TYPE_ERROR = 1
+let s:LSP_MESSAGE_TYPE_WARNING = 2
+let s:LSP_MESSAGE_TYPE_INFORMATION = 3
+let s:LSP_MESSAGE_TYPE_LOG = 4
+
+" Text is going to be echo'ed by ale#util#Execute, so any single quote the
+" text has must be reescaped
+function! s:escapeQuotes(text) abort
+    return substitute(a:text, '''', '''''', 'g')
+endfunction
+
+function! s:echoError(text) abort
+    call ale#util#Execute(
+    \ 'redraw | echohl ErrorMsg | echomsg ''' . s:escapeQuotes(a:text) . ''' | echohl None')
```

Comment:
Remove `redraw`, as I don't think we should force a redraw on every message.

remove the `echohl` commands for now.

I recommend replacing all of these echoes with calls for the `ale#util#ShowMessage` function, which will show larger messages in preview windows. Long messages cannot be echoed in Vim 8 from an asynchronous context.

Reply:
Having a thought about this, I actually wonder what's the best way to display those messages. As a user, I'd expect something non intrusive (e.g. doesn't requires me to press enter for long messages) that keeps the info until I think it's best to look at it, quite similar to `ALEDetail` but keeping old messages. This is food for thought now, I'm happy to hear suggestions.

I'll switch to `ale#util#ShowMessage` and see how that behaves, although that will be skewed since I don't use that many language servers regularly.